Applications/FileIO/SWMM/SWMMInterface.cpp

```diff
@@ -0,0 +1,1248 @@
+        {
+            INFO ("Reading junctions...");
+            if (!readNodeData(in, nodes, name_id_map, max_depth, true))
+                return false;
+        }
+        else if (line == "[OUTFALLS]")
+        {
+            INFO ("Reading outfalls...");
+            if (!readNodeData(in, nodes, name_id_map, max_depth, false))
+                return false;
+        }
+        else if (line == "[STORAGE]")
+        {
+            INFO ("Reading storages...");
+            if (!readNodeData(in, nodes, name_id_map, max_depth, true))
+                return false;
+        }
+        else if (line == "[CONDUITS]")
+        {
+            INFO ("Reading conduits...");
+            if (!readLineElements(in, elements, nodes, name_id_map))
+                return false;
+            n_elem_types[0] = elements.size();
+        }
+        else if (line == "[PUMPS]")
+        {
+            INFO ("Reading pumps...");
+            if (!readLineElements(in, elements, nodes, name_id_map))
+                return false;
+            n_elem_types[1] = elements.size();
+        }
+        else if (line == "[WEIRS]")
+        {
+            INFO ("Reading weirs...")
+            if (!readLineElements(in, elements, nodes, name_id_map))
+                return false;
+            n_elem_types[2] = elements.size();
+        }
+        else if (line == "[POLLUTANTS]")
+        {
+            if (!readPollutants(in))
+                return false;
+        }
+        if (line == "[Polygons]")
+        {
+            INFO ("Reading subcatchments...");
+            std::vector<GeoLib::Polyline*> lines;
+            std::vector<std::string> line_names;
+            std::vector<std::string> tmp_names; // polygon points are nameless but the method requires a vector
+            if (!readPolygons(in, lines, line_names, _subcatchment_points, tmp_names))
+                return false;
+
+            if (!matchSubcatchmentsWithOutlines(lines, line_names))
+                return false;
+        }
+    }
+
+    if (elements.empty())
+    {
+        for (MeshLib::Node* node : nodes)
+            delete node;
+        return false;
+    }
+
+    MeshLib::Properties props;
+    boost::optional< MeshLib::PropertyVector<int>& > mat_ids =
+        props.createNewPropertyVector<int>("MaterialIDs", MeshLib::MeshItemType::Cell, 1);
+    mat_ids->resize(elements.size(), 0);
+    for (std::size_t i=1; i<n_types; ++i)
+        std::fill(mat_ids->begin()+n_elem_types[i-1], mat_ids->begin()+n_elem_types[i], i);
+
+    if (nodes.size() == max_depth.size())
+    {
+        boost::optional< MeshLib::PropertyVector<double>& > depth =
+            props.createNewPropertyVector<double>("Max Depth", MeshLib::MeshItemType::Node, 1);
+        depth->reserve(max_depth.size());
+        std::copy(max_depth.cbegin(), max_depth.cend(), std::back_inserter(*depth));
+    }
+    else
+        ERR ("Size of max depth array does not fit number of elements. Skipping array.");
```

Comment:
Because I'm unsure of the meaning of "max depth". It's not available for all point types and it's input instead of output info. Nothing really points to this being an error, we'd have to consult someone who knows this stuff in detail.